test-clients/src/main/java/com/hedera/services/bdd/spec/util/EncodingUtil.java

```diff
@@ -0,0 +1,124 @@
+        var removeOpeningBracket = argumentTypesToConvert.replace("(", "");
+        var removeClosingBracket = removeOpeningBracket.replace(")", "");
+        final var splittedArgumentTypes = removeClosingBracket.split(",");
+
+        final var convertedArgumentTypes = new String[splittedArgumentTypes.length];
+        for(int i = 0; i < params.length; i++) {
+            if(params[i] instanceof Integer && !("uint32".equals(splittedArgumentTypes[i]))) {
+                convertedArgumentTypes[i] = "int32";
+            } else if((params[i] instanceof Long && !"int64".equals(splittedArgumentTypes[i])) || "int32".equals(splittedArgumentTypes[i])) {
+                convertedArgumentTypes[i] = "int64";
+            } else if(params[i] instanceof BigInteger && !("uint64".equals(splittedArgumentTypes[i])) || "int128".equals(splittedArgumentTypes[i])
+                    || "uint128".equals(splittedArgumentTypes[i]) || "int256".equals(splittedArgumentTypes[i]) || "uint256".equals(splittedArgumentTypes[i])) {
+                convertedArgumentTypes[i] = "uint256";
+            } else {
+                convertedArgumentTypes[i] = splittedArgumentTypes[i];
+            }
+        }
+
+        return "("  + Arrays.stream(convertedArgumentTypes).collect(Collectors.joining(",")).concat(")");
+    }
+
+    private static Object[] convertAddressToHeadlong(final Object[] params) {
+        Object[] result = new Object[params.length];
+        for (int i = 0; i < params.length; i++) {
+            if(params[i] instanceof String && ((String) params[i]).length() > 10){
+                result[i] = convertAliasToHeadlongAddress((String) params[i]);
+            } else if(params[i] instanceof byte[] && (((byte[]) params[i]).length == 20)) {
+                result[i] = asHeadlongAddress((byte[]) params[i]);
+            } else if(params[i] instanceof byte[] && (((byte[]) params[i]).length == 40)) {
+                result[i] = convertAliasToHeadlongAddress((String) params[i]);
+            } else if (params[i] instanceof List) {
+                if(!((List<?>) params[i]).isEmpty() && ((List<?>) params[i]).get(0) instanceof byte[]){
+                    Address[] objectsInside = new Address[((List<?>) params[i]).size()];
+                    for (int j = 0; j < ((List<?>) params[i]).size(); j++) {
+                        if(((byte[]) ((List<?>) params[i]).get(j)).length == 20) {
+                            var instance = ((List<?>) params[i]).get(j);
+                            instance = asHeadlongAddress((byte[]) instance);
+                            objectsInside[j] = (Address) instance;
+                        }
+                        if(((byte[]) ((List<?>) params[i]).get(j)).length == 40) {
+                            var instance = ((List<?>) params[i]).get(j);
+                            instance = convertAliasToHeadlongAddress((String) instance);
+                            objectsInside[j] = (Address) instance;
+                        }
```

Comment:
You can combine the two if blocks in a single block and make the length check nested.

                            ```
var nestedParam = ((List<?>) params[i]).get(j);
                            if(nestedParam != null) {
                              var nestedParamLength = (byte[]) nestedParam).length;
                              if(nestedParamLength == 20) {
                                instance = asHeadlongAddress((byte[]) nestedParam);
                              } else if(nestedParamLength == 40) {
                                instance = convertAliasToHeadlongAddress((String) nestedParam);
                              }
                              objectsInside[j] = (Address) nestedParam;
                            }
```